AGENTS.md

```diff
@@ -0,0 +1,952 @@
+- Verify pytest plugin is loaded in `conftest.py`
+
+```python
+# Check if fixture is in pytest_plugins list
+pytest_plugins = [
+    'pytest_fixtures.component.activationkey',  # Make sure this is loaded
+]
+```
+
+#### 2. **Test Hangs During Execution**
+
+**Problem**: Test hangs indefinitely
+
+**Solution**:
+- Add timeout to long-running operations
+- Use `wait_for()` with proper timeout
+- Check for blocking I/O operations
+
+```python
+from wait_for import wait_for
+
+# Add timeout to wait conditions
+wait_for(
+    lambda: condition_check(),
+    timeout=300,  # 5 minutes
+    delay=10,
+    logger=logger
+)
+```
+
+#### 3. **Broker Checkout Failure**
+
+**Problem**: `BrokerError: No hosts available`
+
+**Solution**:
+- Check Broker configuration in `broker_settings.yaml`
+- Verify inventory has available hosts
+- Check host requirements match available inventory
+
+```bash
+# Check Broker inventory
+broker inventory
+
+# Check specific host requirements
+broker checkout --workflow deploy-rhel --rhel-version 9
+```
+
+#### 4. **Content Host Registration Fails**
+
+**Problem**: Registration fails with certificate errors
+
+**Solution**:
+- Verify Katello CA is installed
+- Check Satellite hostname is resolvable
+- Ensure correct activation key is used
+
+```python
+# Install CA certificate first
+rhel_contenthost.install_katello_ca(target_sat)
+
+# Verify hostname resolution
+result = rhel_contenthost.execute(f'ping -c 1 {target_sat.hostname}')
+assert result.status == 0
+
+# Register with proper parameters
+rhel_contenthost.register_contenthost(
+    org=org,
+    activation_key=ak.name,
+    target=target_sat
+)
+```
+
+#### 5. **UI Test Element Not Found**
+
+**Problem**: `NoSuchElementException` in UI tests
+
+**Solution**:
+- Add `wait_for` before interacting with elements
+- Use `browser.plugin.ensure_page_safe()`
+- Check if element is in an iframe
+
+```python
+from wait_for import wait_for
+
+with target_sat.ui_session() as session:
+    # Wait for page to load
+    wait_for(lambda: session.activationkey.is_displayed, timeout=30)
+    
+    # Ensure page is stable
+    session.browser.plugin.ensure_page_safe(timeout='10s')
+    
+    # Interact with element
+    session.activationkey.create({'name': 'test-ak'})
+```
+
+---
+
+## Development Conventions
+
+### Linting and Code Quality
+
+*   **Linting:** The project uses `ruff` for linting and formatting. The configuration is in `pyproject.toml`.
+    - Line length: 100 characters
+    - Quote style: Preserved (transitioning to single quotes)
+    - Run manually: `ruff check .` or `ruff format .`
+
+*   **Pre-commit Hooks:** The project can use `pre-commit` to run checks before committing.
+    - Run manually: `pre-commit run --all-files`
+
+### Configuration
+
+*   **Test Configuration:** The application uses multiple YAML files for configuration:
+    - `robottelo.yaml`: Main test configuration
+    - `broker_settings.yaml`: Broker/VM configuration
```

Comment:
`broker_settings.yaml` is an optional config that does not need to be present; broker can use `~/.broker/broker_settings.yaml`.